lib/spack/spack/cmd/find.py

```diff
@@ -165,14 +166,23 @@ def setup_parser(subparser):
 
 
 def query_arguments(args):
-    # Set up query arguments.
-    installed = []
-    if not (args.only_missing or args.only_deprecated):
-        installed.append(InstallStatuses.INSTALLED)
-    if (args.deprecated or args.only_deprecated) and not args.only_missing:
-        installed.append(InstallStatuses.DEPRECATED)
-    if (args.missing or args.only_missing) and not args.only_deprecated:
-        installed.append(InstallStatuses.MISSING)
+    if args.only_missing and args.deprecated:
+        raise RuntimeError("cannot use --only-missing with --deprecated")
+
+    if args.only_deprecated and args.missing:
+        raise RuntimeError("cannot use --only-deprecated with --missing")
+
+    installed = StatusQuery.INSTALLED
+    if args.only_missing:
+        installed = StatusQuery.MISSING
+    elif args.only_deprecated:
+        installed = StatusQuery.DEPRECATED
+
+    if args.missing:
+        installed |= StatusQuery.MISSING
```

Comment:
this is wrong for `--only-missing --missing`

Reply:
> Maybe make the whole set --{only-}{missing,deprecated} mutually exclusive then?

If we want `--only-missing --missing` to error (which is fine according to the help, I guess), I can just tweak the initial conditionals:
```python
if args.only_missing and (args.deprecated or args.missing):
    raise RuntimeError("cannot use --only-missing with --deprecated, or --missing")
```
Making them all mutually exclusive prevents `--deprecated --missing` which I think should be valid?

Reply:
The original branches made little sense, as they allowed options together that should have been incompatible:
![Screenshot from 2024-10-30 14-47-05](https://github.com/user-attachments/assets/00b938ef-0d34-4f2a-b0c9-31bd94521d07)

Let me know if https://github.com/spack/spack/pull/47321/commits/10e54fd4cf637b9a5dad8313f358d3d2209f6e9f seems ok